### DISCUSSION TITLE
Window intercepts mouse events but User Control does not

### DISCUSSION THREAD

**Discussion by artemiusgreat** [AvaloniaUI/Avalonia#7047] [Q&A] [answered]
I have `SomeUserControl.axaml.cs` and trying to intercept some mouse events in it. I tried 3 different ways to do it but the handler is not being triggered. At the same time, when I use any of the code snippets in the `MainWindow.axaml.cs` everything works fine and event is being intercepted as expected. 

```C#
public MyCanvasView()
{
  AvaloniaXamlLoader.Load(this);

  // Option A - event delegate
  PointerMoved += (object sender, PointerEventArgs e) => {};

  // Option B - interactive handler
  this.AddHandler(PointerMovedEvent, (object sender, PointerEventArgs e) => {}, RoutingStrategies.Bubble, true);
}

// Option C - override default handler
protected override void OnPointerMoved(PointerEventArgs e)
{
}
```

**Other details** 

https://stackoverflow.com/questions/70115735/mouse-events-are-not-triggered-in-avalonia-user-control 

**Question** 

Is it expected that only `Window` can handle events? Can `UserControl` do it?

**Comment by maxkatz6** [comment, marked as answer]
> Can UserControl do it?

Of course. Same as in WPF.
UserControl is invisible for hit test (pointer input) for obvious reasons to avoid it being blocking input from any other control behind it.
But you can easily make it visible for hit test by making it "visible" - set Background to non-null value. For example, "Transparent".

**Reply by artemiusgreat** [reply]
Thanks. 
Setting background makes it work. 
Meanwhile, WPF controls can intercept these events by default without setting Visibility or Background. 
Here is a sample of the same control in WPF that exposes these events out of the box. 

```
<UserControl
  x:Class="CrossCanvas.Views.WPF.CanvasView"
  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
  xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006" 
  xmlns:d="http://schemas.microsoft.com/expression/blend/2008" 
  xmlns:local="clr-namespace:CrossCanvas.Views.WPF" 
  xmlns:skia="clr-namespace:SkiaSharp.Views.WPF;assembly=SkiaSharp.Views.WPF"
  mc:Ignorable="d" 
  d:DesignHeight="450"
  d:DesignWidth="800"
  SizeChanged="OnSize"
  MouseWheel="OnWheel"
  MouseMove="OnMouseMove" 
  MouseLeave="OnMouseLeave" 
  MouseDoubleClick="OnMouseDown"
  Loaded="OnLoad" 
  Unloaded="OnUnload">
</UserControl>
```

**Reply by maxkatz6** [reply]
I guess it's about default values, can't check now
